crates/api_crud/src/site/read.rs

```diff
@@ -99,8 +109,22 @@ pub async fn get_site(
       discussion_languages,
     })
   } else {
+    // filter oauth_providers for public access
+    filter_oauth_providers(&local_user_view, &mut site_response);
     None
   };
 
   Ok(Json(site_response))
 }
+
+fn filter_oauth_providers(
+  local_user_view: &Option<LocalUserView>,
+  site_response: &mut GetSiteResponse,
+) {
+  if !local_user_view
+    .clone()
```

Comment:
Not good to clone all this data for every request. Use something like `local_user_view.map(|l| l.local_user.admin).unwrap_or_default()`